File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-5.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */

    kthSmallest(root, k) {
    //DO LATER
    if(!root) return;
    let res = [];

    function dfs(root){
        if(!root) return;

         dfs(root.left);
        res.push(root.val);
         dfs(root.right);
    }
    dfs(root);
    return res[k-1];
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        #depth
        levels = {}

        def dfs(depth, node):
            if not node:
                return
            
            if depth in levels:
                levels[depth].append(node.val)
            else:
                levels[depth] = [node.val]
            
            if node.left:
                dfs(depth + 1, node.left)
            if node.right:
                dfs(depth + 1, node.right)
        
        dfs(0, root)
        return list(levels.values())
            